lib/vagrant-libvirt/action/create_domain.rb

```diff
@@ -73,6 +73,33 @@ def call(env)
           @tpm_path = config.tpm_path
           @tpm_version = config.tpm_version
 
+          @sysinfo = {}
+          @sysinfo_blocks = {
+            'bios' => [ 'vendor', 'version', 'date', 'release' ],
+            'system' => [ 'manufacturer', 'product', 'version', 'serial', 'uuid', 'sku', 'family' ],
+            'base_board'=> [ 'manufacturer', 'product', 'version', 'serial', 'asset', 'location' ],
+            'chassis'=> [ 'manufacturer', 'version', 'serial', 'asset', 'sku' ]
+          }
+          @sysinfo_blocks.each do |block_name, valid_entries|
+            if config.sysinfo.has_key?(block_name) and not config.sysinfo[block_name].empty?
+              has_valid_entries = false
+              valid_entries.each do |entry_name|
+                if config.sysinfo[block_name].has_key?(entry_name)
+                  has_valid_entries = true
+                end
+              end
+              if has_valid_entries
+                @sysinfo[block_name] = {}
+              end
+              valid_entries.each do |entry_name|
+                @sysinfo[block_name][entry_name] = config.sysinfo[block_name][entry_name]
+              end
+            end
+          end
+          if config.sysinfo.has_key?('oem_strings') and not config.sysinfo[:oem_strings].empty?
+              @sysinfo['oem_strings'] = config.sysinfo['oem_strings']
+          end
```

Comment:
This seems like effectively validation which should really be done in the config object if it is considered a good idea to filter out invalid entries, and also consider whether to either emit a warning or trigger an error.

Otherwise typos as someone is writing a Vagrantfile cannot be detected by the `vagrant validate` command and mistakes only get picked up when the machine has been brought up.

I appreciate there is probably more churn than you were expecting, so I'm happy to help move stuff around if that's alright with you.

Reply:
I think I addressed this correctly in 31a13ba979fb9952e4f0177ba7db6c4fbb20da80.